### PR TITLE
Show last released date when sorting by relevance

### LINKED ISSUE #5222: Display "last released" date in sorted-by-relevance search results
When the user [searches for, e.g., `short-url`](https://pypi.org/search/?q=short-url), in the default view (sorted by relevance), we could display the last-released date on each item, just as we do [when sorting by last updated](https://pypi.org/search/?q=short-url&o=-created). 

Requested by @koobs in IRC (`#pypa-dev`) today:

> Makes it easier to filter 'relevance results' by last updated for large result sets
> when you search on PyPI, it defaults to a 'Relevance' search. the results in this Relevance (order by) mode dont display 'Last updated' attribute of the package in the results
> When you switch to 'Last updated' order by, 'Last updated' values are shown
> What I'm proposing is exposing/displaying the 'Last updated' values in the 'Relevance' order by case as well
> Allowing users to use the 'relevance' matching which highlights (as expected) 'more' relevant results, but also allows them to see last updated, instead of going to the package page, to see last updated
> my example case: https://pypi.org/search/?q=short+url
> when you switch that search to last updated, most of the results aren't relevant for the search terms any more
<koobs> im looking for 'relevent' results 'updated recently'
<koobs> without 'Last updated' being displayed in relevence mode, i cant do that
~~~

### PR DESCRIPTION
**Pull request by sanjaysiddhanti**
@brainwane cc @di This PR implements https://github.com/pypa/warehouse/issues/5222. It shows the last release date when searching for a package and ordering by relevance.

Here is a screenshot when searching for `requests` and ordering by relevance. 

image

I was initially confused because the dates are so old, and I thought it might be showing package creation date (e.g. date of the first release). However, it looks like I just have old data locally. My local PyPI thinks the most recent `requests` version is 2.5.4.1.

image

Fixes #5222.

### CODE REVIEW

File: warehouse/templates/search/results.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <h3 class="package-snippet__title">
       <span class="package-snippet__name">{{ item.name }}</span>
       <span class="package-snippet__version">{{ item.latest_version }}</span>
-      {% if "created" in order %}
+      {% if "created" in order or order == "" %}
```
Comment (di) [inline warehouse/templates/search/results.html:26]: I think if we're showing this for "Relevance" and "Date last updated" we can probably just show it for "Trending" too and remove this conditional entirely.

Comment (sanjaysiddhanti) [reply warehouse/templates/search/results.html]: Works for me 7464024

image

Comment (di) [reply warehouse/templates/search/results.html]: Unfortunately I think you'll need to `make translations` now that we've removed lines.

**Comment by sanjaysiddhanti** [comment]
I'll fix the build

**Comment by sanjaysiddhanti** [comment]
@di Build is green now

**Comment by di** [comment]
Thanks! FYI for the future: by writing something like `Fixes #123` you can [ensure that the original issue is closed automatically when the PR is merged](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword). I manually added it for you this time.

**Comment by sanjaysiddhanti** [comment]
> by writing something like Fixes #123 you can ensure that the original issue is closed automatically when the PR is merged

Thanks! Will do that next time.

**Pull request merged**